src/target/mod.rs

```diff
@@ -530,6 +530,14 @@ pub trait Target {
         <Self::Arch as Arch>::single_step_gdb_behavior()
     }
 
+    /// Enable/disable `QStartNoAckMode`
```

Comment:
I don't think these docs are up to snuff...
That said, for expediency's sake, I don't mind cleaning these up after merge.

In the future though, docs should avoid mentioning the nitty-gritty details of their underlying implementation whenever possible, and instead, be written from the perspective of user-visible behavior. The point of `gdbstub` is that users really shouldn't care what a `QStartNoAckMode` packet is.

i.e: i'll tweak these docs to paraphrase from https://sourceware.org/gdb/onlinedocs/gdb/Packet-Acknowledgment.html#Packet-Acknowledgment